### PR TITLE
Adds help documentation for errors when generating SDL/Scaffolds with schema relations

### PR DESCRIPTION
**Pull request by dthyresson**
Fix: https://github.com/redwoodjs/redwood/issues/4762
Fix: https://github.com/redwoodjs/redwood/issues/4394
Fix: https://github.com/redwoodjs/redwood/issues/4842

This PR adds helps docs to explain the errors that happen when using Redwood generators with Prisma schema relations in certain circumstances

* add relation between models before each exist
* add a self-relation where the parent isn't optional

Both of these cases create unfriendly errors that we want to make better, but as a workaround until then, these documents can be used to point a developer in the right direction to explain and correct their schema so they can move forward.

### CODE REVIEW

File: docs/docs/schema-relations.md [hunk 1]
```diff
@@ -99,3 +99,155 @@ Which creates a table structure like:
...
+  name  String @unique
+  books Book[]
+}
+```
+
+5. Then **migrate** again to create the relationships in the database
+
+ `yarn rw prisma migrate dev`
+
+ 6. Last, regenerate your SDL (but you will need to `force` to overwrite existing files and use the `no-tests` to preserve your tests and scenario files if needed)
```
Comment (Tobbe) [inline docs/docs/schema-relations.md:217]:
```suggestion
 6. Last, regenerate your SDL (but you will need to `force` to overwrite existing files and use the `--no-tests` flag to preserve your tests and scenario files if needed)
```

File: docs/docs/schema-relations.md [hunk 2]
```diff
@@ -99,3 +99,155 @@ Which creates a table structure like:
...
+
+`yarn rw g sdl Book --force --no-tests`
+`yarn rw g sdl Shelf --force --no-tests`
+
+Now you have a schema, SDL, service that correctly represents your models and relationships.
```
Comment (Tobbe) [inline docs/docs/schema-relations.md:222]:
```suggestion
Now you have a schema, SDL and service that correctly represents your models and relationships.
```

File: docs/docs/schema-relations.md [hunk 3]
```diff
@@ -99,3 +99,155 @@ Which creates a table structure like:
...
+  id            Int       @id @default(autoincrement())
+  name          String
+  jobTitle      String
+  reportsToId   Int?      @unique
+  reportsTo     Employee? @relation("OrgChart", fields: [reportsToId], references: [id])
+  directReports Employee? @relation("OrgChart")
+}
+```
+
+What is important here for the RedwoodJS generators is that the related models **be optional**. That is, the `reportsToId`, `reportsTo`, and `directReports` use Prisma's `?` syntax to indicate that the item is not required.
```
Comment (Tobbe) [inline docs/docs/schema-relations.md:246]:
```suggestion
What is important here for the RedwoodJS generators is that the related models **are optional**. That is, the `reportsToId`, `reportsTo`, and `directReports` use Prisma's `?` syntax to indicate that the item is not required.
```

**Comment by jtoar** [comment]
@dthyresson @Tobbe finally got around to reviewing this one, thanks for the patience!

**Pull request merged**